File: Sort Algorithms/Insertion Sort Algorithm/InsertionSortAlgorithm.cpp
#include <iostream>
#include <vector>

using namespace std;

void insertionSort(vector<int>& nums) {
	for (int i = 0; i < nums.size(); ++i) {
		int j = i;
		while (j > 0 && nums[j] < nums[j - 1]) {
			int temp = nums[j - 1];
			nums[j - 1] = nums[j];
			nums[j] = temp;
			--j;
		}
	}	
}

void print(vector<int>& nums) {
	cout << "[";
	for (int i = 0; i < nums.size(); ++i)
		cout << nums[i] << ((i != nums.size()-1)? ", ":"");
	cout << "]" << endl;
}

int main(int argc, char* argv[]) {
	vector<int> nums = {25, 32, 12, 1, 0, 45, -2, 5};
	print(nums);
	insertionSort(nums);
	print(nums);
	return 0;
}

File: Sort Algorithms/Insertion Sort Algorithm/InsertionSortAlgorithm.py
def insertionSort(nums):
    for i in range(len(nums)):
        j = i
        while j > 0 and nums[j] < nums[j-1]:
            nums[j], nums[j-1] = nums[j-1], nums[j]
            j -= 1

nums = [12, 45, 19, 0, 100, 75, 3, 1]
insertionSort(nums)
print(nums)
